packages/patternfly-3/patternfly-react-extensions/src/components/CatalogTile/CatalogTile.js

```diff
@@ -49,46 +77,61 @@ const CatalogTile = ({
     );
   };
 
-  const truncateDescription = truncateDescriptionFn || defaultTruncateDescription;
-
-  const renderInner = () => (
-    <React.Fragment>
-      <div className="catalog-tile-pf-header">
-        {iconImg && <img className="catalog-tile-pf-icon" src={iconImg} alt="" />}
-        {!iconImg && iconClass && <span className={`catalog-tile-pf-icon ${iconClass}`} />}
-        {renderBadges()}
-      </div>
-      <div className="catalog-tile-pf-body">
-        <div className="catalog-tile-pf-title">{title}</div>
-        <div className="catalog-tile-pf-subtitle">{vendor}</div>
-        <div className="catalog-tile-pf-description">{truncateDescription(description, maxDescriptionLength, id)}</div>
-      </div>
-    </React.Fragment>
-  );
+  render() {
+    const {
+      id,
+      className,
+      featured,
+      href,
+      onClick,
+      iconImg,
+      iconClass,
+      badges,
+      title,
+      vendor,
+      description,
+      truncateDescriptionFn,
+      maxDescriptionLength,
+      ...otherProps
+    } = this.props;
+    const { heightStyle } = this.state;
+    const truncateDescription = truncateDescriptionFn || defaultTruncateDescription;
+
+    const classes = classNames('catalog-tile-pf', { featured }, className);
+
+    const OuterComponent =
+      href || onClick
+        ? ({ children }) => (
+            <a id={id} className={classes} href={href || '#'} {...otherProps} onClick={e => this.handleClick(e)}>
+              {children}
+            </a>
+          )
+        : ({ children }) => (
+            <div id={id} className={classes} {...otherProps}>
+              {children}
+            </div>
+          );
```

Comment:
Notes: could see someone having trouble with the conditional operator, but this works too